src/services/APIService.ts

```diff
@@ -0,0 +1,107 @@
+import camelcaseKeys from 'camelcase-keys';
+import { urlIsAvailable } from '../helpers/urlIsAvailable';
+import { FilterSearch, MovieInfoDetails, MovieResults } from '../types';
+
+const baseUrl = 'http://localhost:4000/';
+
+export default {
+    getMovies: async (offset = 0, limit = 8, filterSearch?: FilterSearch): Promise<MovieResults> => {
+        let url = `${baseUrl}movies?offset=${offset}&limit=${limit}`;
+        if (filterSearch) {
+            const { sortBy, search, sortOrder, searchBy, filter } = filterSearch;
+            if (sortBy) {
+                url += `&sortBy=${sortBy}`;
+                if (sortOrder) {
+                    url += `&sortOrder=${sortOrder}`;
+                } else {
+                    url += `&sortOrder=desc`;
+                }
+            }
+            if (search) {
+                url += `&search=${search}`;
+            }
+            if (searchBy) {
+                url += `&searchBy=${searchBy}`;
+            }
+            if (filter && filter.toString() !== 'All') {
+                url += `&filter=${filter}`;
+            }
+        }
+        const response = await fetch(`${url}`);
```

Comment:
fetch(url)?

Reply:
yes, thanks. I updated this in commit e9860952613603eb2cf3095f28c180d207b9668c